File: PHP/merged-pdf-endpoint.php
<?php
require("../Sample_Input/sample_input.php");

// The /merged-pdf endpoint can take one or more PDF files or ids as input.
// This sample takes 2 PDF files and merges all the pages in the document into a single document.
$merged_pdf_endpoint_url = 'https://cloud-api.datalogics.com/merged-pdf';

// Create an array that contains that data that will be passed to the POST request.
$data = array(
    'file' => array(
        SAMPLE_INPUT_DIR . 'merge1.pdf',
        SAMPLE_INPUT_DIR . 'merge2.pdf'
    ),
    'pages' => array("1-last", "1-last"),
    'type' => array ('file', 'file'),
    'output' => 'example_mergedPdf_out'
);

$headers = array(
    'Accept: application/json',
    'Content-Type: multipart/form-data',
    'Api-Key: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' // place your api key here
);

// Form cURL POST command that will be executed with the merged-pdf endpoint.
// NOTE: The '-s' in the cURL command below runs cURL in silent mode, so exec() output is not shown.
$curl_command = 'curl -s -X POST "'.$merged_pdf_endpoint_url.'" -H "'.$headers[0].'" -H "'.$headers[1].'" -H "'.$headers[2].'" -F "file=@'.$data['file'][0].'" -F "pages[]='.$data['pages'][0].'" -F "type[]='.$data['type'][0].'" -F "file=@'.$data['file'][1].'" -F "pages[]='.$data['pages'][1].'" -F "type[]='.$data['type'][1].'" -F "output='.$data['output'].'"';

print "Sending POST request to merged-pdf endpoint...\n";
exec($curl_command, $response);

print json_encode(json_decode($response[0]), JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
print "\n";
?>

File: DotNet/merged-pdf-endpoint.cs
using System.Text;

using (var httpClient = new HttpClient { BaseAddress = new Uri("https://api.pdfrest.com") })
{
    using (var request = new HttpRequestMessage(HttpMethod.Post, "merged-pdf"))
    {
        request.Headers.TryAddWithoutValidation("Api-Key", "XXXXXXXXXXXX");
        request.Headers.Accept.Add(new("application/json"));
        var multipartContent = new MultipartFormDataContent();

        var byteArray = File.ReadAllBytes("/path/to/file1.pdf");
        var byteAryContent = new ByteArrayContent(byteArray);
        multipartContent.Add(byteAryContent, "file", "file1.pdf");
        byteAryContent.Headers.TryAddWithoutValidation("Content-Type", "application/pdf");

        var byteArrayOption = new ByteArrayContent(Encoding.UTF8.GetBytes("file"));
        multipartContent.Add(byteArrayOption, "type[]");
        var byteArrayOption2 = new ByteArrayContent(Encoding.UTF8.GetBytes("all"));
        multipartContent.Add(byteArrayOption2, "pages[]");


        var byteArray2 = File.ReadAllBytes("/path/to/file2.pdf");
        var byteAryContent2 = new ByteArrayContent(byteArray2);
        multipartContent.Add(byteAryContent2, "file", "file2.pdf");
        byteAryContent2.Headers.TryAddWithoutValidation("Content-Type", "application/pdf");

        var byteArrayOption3 = new ByteArrayContent(Encoding.UTF8.GetBytes("file"));
        multipartContent.Add(byteArrayOption3, "type[]");
        var byteArrayOption4 = new ByteArrayContent(Encoding.UTF8.GetBytes("all"));
        multipartContent.Add(byteArrayOption4, "pages[]");

        request.Content = multipartContent;
        var response = await httpClient.SendAsync(request);

        var apiResult = await response.Content.ReadAsStringAsync();

        Console.WriteLine("API response received.");
        Console.WriteLine(apiResult);
    }
}
